### DISCUSSION TITLE
What is the right way to do multiple transformation using open3d?

### DISCUSSION THREAD

**Discussion by li-chengyi** [isl-org/Open3D#6191] [Q&A]
Hi everybody, I have some trouble with the `transform` method of `o3d.geometry.PointCloud`:
I found that I get two different outcome when doing multiple transformation using `transform` method  and matrix multiply.
matrix multiply (`pcd` is a Nx4 matrix):

        pcd = pcd.T
        pcd = np.dot(trans_f2w, np.dot(rot_ff, np.dot(shift_tr, pcd)))
        pcd = pcd.T
        pcd_vector = o3d.geometry.PointCloud()
        pcd_vector.points = o3d.utility.Vector3dVector(pcd[:, :3])

 `transform` method:

        pcd_vector = o3d.geometry.PointCloud()
        pcd_vector.points = o3d.utility.Vector3dVector(pcd[:, :3])
        pcd_vector.transform(shift_tr)
        pcd_vector.transform(rot_ff)
        pcd_vector.transform((trans_f2w))

What is the profound mystery inside `transform` ?

**Comment by ssheorey** [comment]
Your two pieces of code should give the same result. Please file a bug report with an example if it doesn't. Transform does left multiplication and each column of x is a homogenous point:

`y2 = T2 @ T1 @ x .... `

is the same as

```
pcd_vector.transform(T1)
pcd_vector.transform(T2)
```


https://github.com/isl-org/Open3D/blob/f8fbef1d87973f6558f9bf1f940d58d58fbf8391/cpp/open3d/geometry/Geometry3D.cpp#L77
